spawn_operator/spawn_operator/lib/spawn_operator/k8s/proxy/deployment.ex

```diff
@@ -228,25 +232,39 @@ defmodule SpawnOperator.K8s.Proxy.Deployment do
       ]
     }
 
-    actor_host_function_ports = Map.get(host_params, "ports", [])
-
-    host_container = %{
-      "name" => "actor-host-function",
-      "image" => actor_host_function_image,
-      "env" => actor_host_function_envs,
-      "resources" => actor_host_function_resources
-    }
-
     host_container =
-      if length(actor_host_function_ports) > 0 do
-        Map.put(host_container, "ports", actor_host_function_ports)
-      else
-        host_container
-      end
+      %{
+        "name" => "actor-host-function",
+        "image" => actor_host_function_image,
+        "env" => actor_host_function_envs,
+        "resources" => actor_host_function_resources
+      }
+      |> maybe_put_ports_to_host_container(host_params)
+      |> maybe_put_volume_mounts_to_host_container(host_params)
 
     [
       proxy_container,
       host_container
     ]
   end
+
+  defp maybe_put_ports_to_host_container(result, host_params) do
+    actor_host_function_ports = Map.get(host_params, "ports", [])
+
+    if length(actor_host_function_ports) > 0 do
+      Map.put(result, "ports", actor_host_function_ports)
+    else
+      result
+    end
+  end
+
+  defp maybe_put_volume_mounts_to_host_container(result, host_params) do
+    actor_host_volume_mounts = Map.get(host_params, "volumeMounts", [])
+
+    if length(actor_host_volume_mounts) > 0 do
+      Map.put(result, "volumeMounts", actor_host_volume_mounts)
```

Comment:
I didn't see where volume is added. Adding only the volumeMounts will not work as it is necessary to map the mount point directly to the "physical" volume.

Ref https://kubernetes.io/docs/concepts/storage/volumes/